### PR TITLE
fix: remove stuttering on talk

### PR DESCRIPTION
**Pull request by samulopez**

### CODE REVIEW

File: UiTalkAsyncHandler.cs [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+        if (this.configuration.TranslateNpcNames && nameNode != null && !nameNode->NodeText.IsEmpty)
+        {
+          nameNode->SetText(this.translatedName);
+        }
+
+        var parentNode = talkAddon->GetNodeById(10);
+        textNode->TextFlags = (byte)(TextFlags)((byte)TextFlags.WordWrap | (byte)TextFlags.MultiLine);
+        textNode->SetWidth(parentNode->GetWidth());
+        textNode->SetText(this.translatedText);
+        textNode->ResizeNodeForCurrentText();
```
Comment (samulopez) [inline UiTalkAsyncHandler.cs:108]: using the width of the parent the resizing works

File: UiTalkAsyncHandler.cs [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+              {
+                this.currentNameTranslationId = Environment.TickCount;
+                this.currentNameTranslation = Resources.WaitingForTranslation;
+                int nameId = this.currentNameTranslationId;
+                string nameTranslation = this.Translate(nameToTranslate);
+                this.nameTranslationSemaphore.Wait();
+                if (nameId == this.currentNameTranslationId)
+                {
+                  this.currentNameTranslation = nameTranslation;
+                  // TODO: Implement ImGuiScene for Talk
```
Comment (samulopez) [inline UiTalkAsyncHandler.cs:258]: TODO

File: UiTalkAsyncHandler.cs [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+      if (!this.configuration.TranslateTalk)
+      {
+        return;
+      }
+
+      if (args is not AddonRefreshArgs refreshArgs)
+      {
+        this.TranslateTalkReplacing();
+        return;
+      }
```
Comment (samulopez) [inline UiTalkAsyncHandler.cs:365]: We translate on the refresh event without blocking. We replace on the draw event

File: UiTalkAsyncHandler.cs (hunk 1 above)
Comment (lokinmodar) [reply UiTalkAsyncHandler.cs:108]: Nice find!!

File: UiTalkAsyncHandler.cs (hunk 2 above)
Comment (lokinmodar) [reply UiTalkAsyncHandler.cs:258]: I'll handle this :)

File: UiTalkAsyncHandler.cs (hunk 3 above)
Comment (lokinmodar) [reply UiTalkAsyncHandler.cs:365]: Nice! I was testing which event sequence was the best yesterday.

**Review by lokinmodar — APPROVED** [review]
Nice! I'll add the ImGui part and release this to allow for ppl toplay the new expac with the translations and slowly port the logic to the neutral handler

**Pull request merged**